monkey/tests/unit_tests/common/configuration/test_agent_configuration.py

```diff
@@ -169,12 +171,60 @@ def test_agent_configuration():
     assert isinstance(config.credential_collectors[0], PluginConfiguration)
     assert isinstance(config.payloads[0], PluginConfiguration)
     assert isinstance(config.propagation, PropagationConfiguration)
-    assert config_dict == AGENT_CONFIGURATION
+    assert json.loads(config_json) == AGENT_CONFIGURATION
+
+
+def test_incorrect_type():
+    valid_config = AgentConfiguration.from_mapping(AGENT_CONFIGURATION)
+    with pytest.raises(InvalidConfigurationError):
+        valid_config_dict = valid_config.__dict__
+        valid_config_dict["keep_tunnel_open_time"] = "not_a_float"
+        AgentConfiguration(**valid_config_dict)
 
 
 def test_default_agent_configuration():
+    config = AgentConfiguration.from_json(DEFAULT_AGENT_CONFIGURATION_JSON)
+
+    assert isinstance(config, AgentConfiguration)
+
+
+def test_from_dict():
     schema = AgentConfigurationSchema()
+    dict_ = json.loads(DEFAULT_AGENT_CONFIGURATION_JSON)
 
-    config = schema.loads(DEFAULT_AGENT_CONFIGURATION_JSON)
+    config = AgentConfiguration.from_mapping(dict_)
 
-    assert isinstance(config, AgentConfiguration)
+    assert schema.dump(config) == dict_
+
+
+def test_from_dict__invalid_data():
+    dict_ = json.loads(DEFAULT_AGENT_CONFIGURATION_JSON)
+    dict_["payloads"] = "payloads"
+
+    with pytest.raises(InvalidConfigurationError):
+        AgentConfiguration.from_mapping(dict_)
+
+
+def test_from_json():
+    schema = AgentConfigurationSchema()
+    dict_ = json.loads(DEFAULT_AGENT_CONFIGURATION_JSON)
+
+    config = AgentConfiguration.from_json(DEFAULT_AGENT_CONFIGURATION_JSON)
+
+    assert schema.dump(config) == dict_
+
+
+def test_from_json__invalid_data():
+    invalid_dict = json.loads(DEFAULT_AGENT_CONFIGURATION_JSON)
+    invalid_dict["payloads"] = "payloads"
+
+    with pytest.raises(InvalidConfigurationError):
+        AgentConfiguration.from_json(json.dumps(invalid_dict))
+
+
+def test_to_json():
```

Comment:
Maybe it's worth considering merging the serialization tests. Not only because `from_json` and `to_json` will be used in tandem, but also to avoid the maintenance of default configuration in json format

Reply:
I added `DEFAULT_AGENT_CONFIGURATION` and removed `DEFAULT_AGENT_CONFIGURATION_JSON`